drake/systems/plants/RigidBodySystem.h

```diff
@@ -130,7 +130,7 @@ class LoopConstraint : public LinearEqualityConstraint {
  * with additional sensors and actuators/forces
  * @concept{system_concept}
  */
-class DRAKERBSYSTEM_EXPORT RigidBodySystem {
+class DRAKERBSYSTEM_EXPORT RigidBodySystem : public std::enable_shared_from_this<RigidBodySystem> {
```

Comment:
@RussTedrake I had to add this to get rid of all the raw pointers being passed around.  As a consequence, the RigidBodySystem constructor now takes no arguments and adding URDFs is done as a second step.  I noticed that `enabled_shared_from_this` is no longer present in the rest of the code base.  Are we moving away from this?  It could also be done with references.